src/torchmetrics/classification/precision_recall_curve.py

```diff
@@ -104,9 +129,25 @@ class BinaryPrecisionRecallCurve(Metric):
     higher_is_better: Optional[bool] = None
     full_state_update: bool = False
 
+    class _ComputationMode(Enum):
+        """Internal state of the dynamic mode."""
+
+        BINNED = "binned"
+        NON_BINNED = "non-binned"
+        NON_BINNED_DYNAMIC = "non-binned-dynamic"
```

Comment:
it seems weird to me having a class inside a class def?

Reply:
yes please